### PR TITLE
Df/ky send email function

### LINKED ISSUE #31: Email sender - functions to send email
In this ticket we'll be creating the functions that send the email's we've parsed from the CSV. We will be using the Mailgun API! See a simple example of how to use the API [here](https://documentation.mailgun.com/en/latest/quickstart-sending.html#how-to-start-sending-email) (be sure to click on the Node.js tab on the top, and look at "Send via API").

**---You will not be connecting this to the email sender page yet!**

First add two new variables to `.env` - the mailgun API key (ask @jzhzhang66 for this), and the email domain - `mailgun.hackbeanpot.com`.

You will write one function:
1, A function to send the actual emails to recipients. This function will take in a list of `Message` objects (defined in `lib/types.ts`), and will send each message in a batch. See mailgun's batch functionality [here](https://documentation.mailgun.com/en/latest/user_manual.html#batch-sending-1).

You can test this function locally by adding a temporary button to `pages/emailSender.tsx` - note that the UI for that page isn't finished yet.

In addition to all of this, investigate whether Mailgun includes email signatures when it sends emails.

### PR DESCRIPTION
**Pull request by karynayen**
closes #31 
This PR includes the following changes:
- adds the ability to send batch emails, meaning all contacts listed in the CSV will be emailed at once
- email sending functionality can be tested by pressing the temporary 'send emails' button at the top of the page. (UI will be updated in the future)

https://user-images.githubusercontent.com/37753574/177025650-479d0565-7940-46ca-902a-e9622df99fc3.mov

### CODE REVIEW

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
OMGGG this is soooo exciting!!! The sending works great for me!!

The only thing is that our email sender message depends on what the user passes in as the message and CSV. Although this works great with the fields name + company, it's possible we won't have those fields for some emails or we will have additional fields like school, city, years, etc. or any other possible things that we want to change across different emails. 

If I pass in another column like 'school' it'll print this on the front end
Screen Shot 2022-07-03 at 3 42 01 PM

but send this where ${school} is not replaced
Screen Shot 2022-07-03 at 3 41 35 PM

So I guess I was wondering why instead of passing in the csv data (name, company, etc) why we can't just pass in the email body that we've printed out on the front end? Also since if we're doing the regex / creating the email body in two different places wouldn't that mean what the user sees as printed on the front end might be different from what's sent? 

But yeah do ya'll think it would make sense to have something like

export interface EmailCsvData {
  email: string,
  subject: string,
  bodyMessage: string
}

and not pass in the emailText so we have one source of truth?

**Review by jhylisa666 — CHANGES_REQUESTED** [review]
omg yessss great job!!! This is such a hard ticket oml. 
I agree with Judy that we shoudn't hardcode the EmailCsvData part in case there are other variables like school. But otherwise GREAT JOB!!!!!

**Comment by deanframe** [comment]
@jzhzhang66 I initially thought that we would have to do the parsing on the backend as well because of how Mailgun wants batch messages formatted, but your comments made me think of a better way to do it - will work on it and hopefully request your review again tomorrow!!

File: pages/emailSender.tsx [hunk 1]
```diff
@@ -124,30 +124,30 @@ const EmailSender: NextPage = () => {
...
-      resultMessage: {
-        isError: localErrorMessages.length > 0,
-        message:
-          localErrorMessages.length > 0
-            ? `Error sending ${localErrorMessages.length} of ${finalMessages.length}`
-            : 'Sent emails successfully'
-      }
-    })
-  }
+  // const sendEmails = () => {
```
Comment (deanframe) [inline pages/emailSender.tsx:127]: unsure if we even need local error messages - i'll work on making our error checking system better next week if that works

Comment (jzhzhang66) [reply pages/emailSender.tsx:127]: Sounds good!

File: pages/emailSender.tsx [hunk 2]
```diff
@@ -208,10 +208,42 @@ const EmailSender: NextPage = () => {
...
+        // TODO: surface this to UI
+        console.log(data.result)
+      })
+  }
+
   return (
     <Layout>
       <ThemeProvider theme={theme}>
         <StyledPageContainer>
+          <Button onClick={sendEmails}>Send emails!</Button>
```
Comment (jzhzhang66) [inline pages/emailSender.tsx:246]: Let's remove this old button now that we have the styled one

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Sorry one more thing - let's remove the old button and then I'm good to approve!

**Review by jhylisa666 — APPROVED** [review]
slayyyy

**Pull request merged**